agent/lib/kontena/models/service_pod.rb

```diff
@@ -349,6 +356,10 @@ def build_env
         secrets_hash.each do |name, value|
           env << "#{name}=#{value}"
         end
+        env.each do |envstr|
+          name, value = envstr.split('=', 2)
+          raise ConfigError, "Env #{name} is too large at #{envstr.bytesize} bytes" if envstr.bytesize > ENV_MAX_STRLEN
```

Comment:
No idea how `envstr.length` vs `envstr.bytesize` really works here in case of non-ASCII env values. Maybe if everything is in UTF-8, from the `envstr.encoding` through the docker-api gem client and Docker itself, then this will work correctly?

Reply:
There may also be some odd edge cases here for envs that are exactly 128KB in size - like does the kernel limit include the terminating NUL or some other overhead? Have not tested for those edgecases yet. Maybe the validation should just be a couple bytes stricter just to avoid those?